### PR TITLE
Posting new code from Jingjing Li for the Batch .NET quickstart

### PR DESCRIPTION
**Pull request by v-thepet**
## Purpose

Updated code for the Batch .NET quickstart https://learn.microsoft.com/azure/batch/quick-run-dotnet

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
Updated C# app

```
[x ] Bugfix (GitHub issue)
[ ] Feature
[x ] Code style update (formatting, local variables)
[x ] Refactoring (no functional changes, no api changes)
[x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Build and run the program

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by v-thepet** [comment]
#sign-off - whoever can merge thank you to clear GitHub issue https://github.com/MicrosoftDocs/azure-docs/issues/92899

**Review by Padmalathas — APPROVED** [review]

**Comment by Padmalathas** [comment]
#sign-off

**Comment by Padmalathas** [comment]
@dlepow, Request you to please approve the changes so that its auto-merged to the repo. Thanks!

File: README.md [hunk 1]
```diff
@@ -12,12 +12,12 @@ urlFragment: batch-dotnet-quickstart
...
 A basic .NET application that introduces Batch features such as pools, nodes, jobs, tasks, and interaction with Storage. Each task writes a small text file to standard output.
 
-For details and explanation, see the accompanying article [Run your first Batch job with the .NET API](https://docs.microsoft.com/azure/batch/quick-run-dotnet).
+For details and explanation, see the accompanying article [Quickstart: Use .NET to create a pool and run a job](https://docs.microsoft.com/azure/batch/quick-run-dotnet).
 
 ## Prerequisites
 
 - Azure Batch account and linked general-purpose Azure Storage account
-- Visual Studio 2019, or [.NET Core 2.1](https://www.microsoft.com/net/download/dotnet-core/2.1) for Linux, macOS, or Windows
+- Visual Studio 2019, or [.NET Core 6.0](https://www.microsoft.com/net/download/dotnet) for Linux or Windows
```
Comment (dpwatrous) [inline README.md:20]: Should we make this "Visual Studio 2019 or higher" and ".NET 6.0 or higher"? (.NET 6 is no longer called ".NET Core")

**Review by dpwatrous — CHANGES_REQUESTED** [review]
This looks really good overall, just a few minor comments.

File: BatchDotnetQuickstart/Program.cs [hunk 2]
```diff
@@ -252,55 +233,64 @@ private static ImageReference CreateImageReference()
...
+                    sasBuilder.Identifier = storedPolicyName;
+                }
 
-            return ResourceFile.FromUrl(blobSasUri, filePath);
+                var sasUri = blobClient.GenerateSasUri(sasBuilder).ToString();
+                return ResourceFile.FromUrl(sasUri, filePath);
+            }
+            else
+            {
+                Console.WriteLine(@"BlobClient must be authorized with Shared Key
```
[the remark this answers is not in the document]
Comment (v-thepet) [reply BatchDotnetQuickstart/Program.cs]: @dpwatrous How is this? Please request any changes or merge if ok! Thanks

**Comment by v-thepet** [comment]
@dpwatrous Please critique the try catch block I have added and let me know any further changes or please merge if ok! Thanks
@Padmalathas FYI

**Review by dpwatrous — APPROVED** [review]
The changes look good to me - thanks for the quick update! I'll go ahead and merge it.

**Pull request merged**